File: source/python/cli_tool.py
tool_name = "hemlock-tool"
version = "1.0.0"

config = {
    "verbose": False,
    "output": "stdout",
    "format": "text",
    "count": 10,
}

def parse_option(key, value, cfg):
    if key in ("--verbose", "-v"):
        cfg["verbose"] = True
    elif key in ("--output", "-o"):
        cfg["output"] = value
    elif key in ("--format", "-f"):
        cfg["format"] = value
    elif key in ("--count", "-n"):
        cfg["count"] = int(value)
    return cfg

simulated_args = [
    ("--verbose", ""),
    ("--output", "result.txt"),
    ("--format", "json"),
    ("--count", "25"),
]

for key, val in simulated_args:
    config = parse_option(key, val, config)

print(f"{tool_name} v{version}")
print("Parsed configuration:")
print(f"  verbose: {str(config['verbose']).lower()}")
print(f"  output:  {config['output']}")
print(f"  format:  {config['format']}")
print(f"  count:   {config['count']}")
print("Ready to process.")

File: source/javascript/cli_tool.js
// CLI tool with argument parsing (using defaults)
const toolName = "hemlock-tool";
const version = "1.0.0";

let config = { verbose: false, output: "stdout", format: "text", count: 10 };

function parseOption(key, value, cfg) {
    if (key === "--verbose" || key === "-v") cfg.verbose = true;
    else if (key === "--output" || key === "-o") cfg.output = value;
    else if (key === "--format" || key === "-f") cfg.format = value;
    else if (key === "--count" || key === "-n") cfg.count = parseInt(value);
    return cfg;
}

const simulatedArgs = [
    ["--verbose", ""],
    ["--output", "result.txt"],
    ["--format", "json"],
    ["--count", "25"]
];

for (const arg of simulatedArgs) config = parseOption(arg[0], arg[1], config);

console.log(`${toolName} v${version}`);
console.log("Parsed configuration:");
console.log(`  verbose: ${config.verbose}`);
console.log(`  output:  ${config.output}`);
console.log(`  format:  ${config.format}`);
console.log(`  count:   ${config.count}`);
console.log("Ready to process.");
